mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -1,64 +1,307 @@
-			default:
-				rName = rState.Primary.ID
+			log.Printf("[DEBUG]TGC Terraform metadata: %s", encodedData)
+		}
+
+		return nil
+	}
+}
+
+// parseResources extracts all resources from a Terraform configuration string
+func parseResources(config string) []string {
+	// This regex matches resource blocks in Terraform configurations
+	resourceRegex := regexp.MustCompile(`resource\s+"([^"]+)"\s+"([^"]+)"`)
+	matches := resourceRegex.FindAllStringSubmatch(config, -1)
+
+	var resources []string
+	for _, match := range matches {
+		if len(match) >= 3 {
+			// Combine resource type and name to form the address
+			resources = append(resources, fmt.Sprintf("%s.%s", match[1], match[2]))
+		}
+	}
+
+	return resources
+}
+
+// getServicePackage determines the service package for a resource type
+func getServicePackage(resourceType string) string {
+	var ServicePackages = map[string]string{
+		"google_compute_":   "compute",
+		"google_storage_":   "storage",
+		"google_sql_":       "sql",
+		"google_container_": "container",
+		"google_bigquery_":  "bigquery",
+		"google_project":    "resourcemanager",
+		"google_cloud_run_": "cloudrun",
+	}
+
+	// Check for exact matches first
+	if service, ok := ServicePackages[resourceType]; ok {
+		return service
+	}
+
+	// Check for prefix matches
+	for prefix, service := range ServicePackages {
+		if strings.HasPrefix(resourceType, prefix) {
+			return service
+		}
+	}
+
+	// Default to "unknown" if no match found
+	return "unknown"
+}
+
+// determineImportMetadata checks if the next step is an import step and extracts all import metadata
+func determineImportMetadata(steps []resource.TestStep, currentStepIndex int, resourceName string) ImportMetadata {
+	var metadata ImportMetadata
+
+	// Check if there's a next step and if it's an import step
+	if currentStepIndex+1 < len(steps) {
+		nextStep := steps[currentStepIndex+1]
+
+		// Check if it's an import step for our resource
+		if nextStep.ImportState && (nextStep.ResourceName == resourceName ||
+			strings.HasSuffix(nextStep.ResourceName, "."+strings.Split(resourceName, ".")[1])) {
+			// Capture ignored fields if present
+			if nextStep.ImportStateVerify && len(nextStep.ImportStateVerifyIgnore) > 0 {
+				metadata.IgnoredFields = nextStep.ImportStateVerifyIgnore
+			}
+
+			// If ImportStateId is explicitly set, use that
+			if nextStep.ImportStateId != "" {
+				metadata.Id = nextStep.ImportStateId
+				return metadata
+			}
+
+			// If ImportStateIdPrefix is set, note it
+			if nextStep.ImportStateIdPrefix != "" {
+				metadata.Id = fmt.Sprintf("%s<AUTO_ID>", nextStep.ImportStateIdPrefix)
+				return metadata
 			}
-			caiAssetName := fmt.Sprintf("//%s/%s", apiServiceName, rName)
-			log.Printf("[DEBUG]TGC CAI asset names start\n%s\nEnd of TGC CAI asset names", caiAssetName)
+
+			// If ImportStateIdFunc is set, get function info
+			if nextStep.ImportStateIdFunc != nil {
+				metadata.Id = "<DYNAMIC_IMPORT_ID>"
```

Comment:
I don't see `<DYNAMIC_IMPORT_ID>` is replaced anywhere.

Reply:
DYNAMIC_IMPORT_ID is stating that a function is used to supply the import id. There are about 3 or 4 instances of this. Unfortunately I dont see a way to supply the was import function at this time. Dynamic is intended.

Reply:
hmm actually we might be able to send terraform state to the function directly

Reply:
But I would prefer to do this in another pr